torch/_prims_common/__init__.py

```
@@ -362,6 +432,25 @@ def is_contiguous_for_memory_format(  # type: ignore[return]
     )
 
 
+# similar to is_contiguous_for_memory_format but return false on data dependency.
+def is_known_contiguous_for_memory_format(  # type: ignore[return]
+    a: Tensor, *, memory_format: torch.memory_format
+) -> bool:
+    validate_memory_format(memory_format)
+
+    if memory_format == torch.contiguous_format:
+        return is_known_contiguous(a)
+    if memory_format == torch.channels_last:
+        return is_known_channels_last_contiguous_2d(a)
+    if memory_format == torch.channels_last_3d:
+        return is_known_channels_last_contiguous_3d(a)
+
+    torch._check(
+        False,
+        lambda: f"is_contiguous received unsupported memory format {memory_format}",
+    )
```

would it be better to raise an exception instead of torch._check(false)?

---

mm i just copied 
```
def is_contiguous_for_memory_format(  # type: ignore[return]
    a: Tensor, *, memory_format: torch.memory_format
) -> bool:
    validate_memory_format(memory_format)
    if memory_format == torch.contiguous_format:
        return is_contiguous(a)
    if memory_format == torch.channels_last:
        return is_channels_last_contiguous_2d(a)
    if memory_format == torch.channels_last_3d:
        return is_channels_last_contiguous_3d(a)
    torch._check(
        False,
        lambda: f"is_contiguous received unsupported memory format {memory_format}",
    )
    ```
    I will do a refatctor as mentioned above to pass a flag instead of replicating the function

---

this is existing code, with the new change it will show more appropriately